Sources/Quick/World.swift

```diff
@@ -152,6 +152,28 @@ final internal class World: NSObject {
     internal var exampleCount: Int {
         return allExamples.count
     }
+    
+    internal func beforesCurrentlyExecuting() -> Bool {
+        let suiteBeforesExecuting = (suiteHooks.beforesStartedExecuting && !(suiteHooks.beforesAlreadyExecuted))
+        let exampleBeforesExecuting = (exampleHooks.beforesStartedExecuting && !(exampleHooks.beforesAlreadyExecuted))
+        var groupBeforesExecuting = false
+        if let runningExampleGroup = currentExampleMetadata?.example.group {
+            groupBeforesExecuting = (runningExampleGroup.beforesStartedExecuting && !(runningExampleGroup.beforesAlreadyExecuted))
+        }
+        
+        return suiteBeforesExecuting || exampleBeforesExecuting || groupBeforesExecuting
+    }
+    
+    internal func aftersCurrentlyExecuting() -> Bool {
+        let suiteAftersExecuting = (suiteHooks.aftersStartedExecuting && !(suiteHooks.aftersAlreadyExecuted))
+        let exampleAftersExecuting = (exampleHooks.aftersStartedExecuting && !(exampleHooks.aftersAlreadyExecuted))
+        var groupAftersExecuting = false
+        if let runningExampleGroup = currentExampleMetadata?.example.group {
+            groupAftersExecuting = (runningExampleGroup.aftersStartedExecuting && !(runningExampleGroup.aftersAlreadyExecuted))
+        }
+        
+        return suiteAftersExecuting || exampleAftersExecuting || groupAftersExecuting
+    }
```

Comment:
nit: Would these be better as derived variables? That is:

```swift
internal var beforesCurrentlyExecuting: Bool {
  // ...
  return suiteBeforesExecuting || exampleBeforesExecuting || groupBeforesExecuting
}
```

I'd be fine either way, but vars would be more consistent with the other variables, like `aftersStartedExecuting`.